src/x509_str.c

```diff
@@ -422,7 +436,20 @@ static int X509StoreVerifyCert(WOLFSSL_X509_STORE_CTX* ctx)
                                                         WOLFSSL_SUCCESS : ret;
     #endif
     }
-
+    #if !defined(NO_ASN_TIME) && (defined(OPENSSL_ALL) || defined(WOLFSSL_QT))
+        if (ret != WC_NO_ERR_TRACE(ASN_BEFORE_DATE_E) &&
+            ret != WC_NO_ERR_TRACE(ASN_AFTER_DATE_E)) {
+        /* With Qt and OpenSSL, we need to check the certificate's date
+         * after certificate manager verification,
+         * as it skips date validation when other errors are present.
+         */
+            ret = X509StoreVerifyCertDate(ctx, ret);
+            SetupStoreCtxError(ctx, ret);
+            if (ctx->store->verify_cb)
+                ret = ctx->store->verify_cb(ret >= 0 ? 1 : 0,
+                                            ctx) == 1 ? WOLFSSL_SUCCESS : -1;
```

Comment:
This is much more readable.
```suggestion
                ret = ret == 0 ? 1 : 0
                if (ctx->store->verify_cb(ret, ctx) == 1)
                    ret = WOLFSSL_SUCCESS;
                else
                    ret = -1;
```